src/iter/mod.rs

```diff
@@ -41,16 +41,16 @@ impl<T: BitSetLike> BitIter<T> {
     }
 }
 
-impl<'a> BitIter<&'a mut BitSet> {
+impl<'a, T: UnsignedInteger> BitIter<&'a mut GenericBitSet<T>> {
     /// Clears the rest of the bitset starting from the next inner layer.
     pub(crate) fn clear(&mut self) {
         use self::State::Continue;
         while let Some(level) = (1..LAYERS).find(|&level| self.handle_level(level) == Continue) {
             let lower = level - 1;
-            let idx = (self.prefix[lower] >> BITS) as usize;
-            *self.set.layer_mut(lower, idx) = 0;
+            let idx = (self.prefix[lower] >> T::BITS) as usize;
+            *self.set.layer_mut(lower, idx) = T::ZERO;
             if level == LAYERS - 1 {
-                self.set.layer3 &= !((2 << idx) - 1);
+                self.set.layer3 &= T::from_u32(!((2usize << idx) - 1) as u32);
```

Comment:
Is `u32` enough?

Reply:
Actually I thought the original code was base on a `u32` in this line but this is wrong: it was a `usize`. I'll change to converting to hardcoded `u64` before going back to `T` to ensure it's enough